src/RuleSet/DeclarationBlock.php

```diff
@@ -338,20 +338,21 @@ public function expandFontShorthand(): void
             if (!$mValue instanceof Value) {
                 $mValue = mb_strtolower($mValue);
             }
-            if (in_array($mValue, ['normal', 'inherit'])) {
+            if (in_array($mValue, ['normal', 'inherit'], true)) {
                 foreach (['font-style', 'font-weight', 'font-variant'] as $sProperty) {
                     if (!isset($aFontProperties[$sProperty])) {
                         $aFontProperties[$sProperty] = $mValue;
                     }
                 }
-            } elseif (in_array($mValue, ['italic', 'oblique'])) {
+            } elseif (in_array($mValue, ['italic', 'oblique'], true)) {
                 $aFontProperties['font-style'] = $mValue;
             } elseif ($mValue == 'small-caps') {
                 $aFontProperties['font-variant'] = $mValue;
             } elseif (
-                in_array($mValue, ['bold', 'bolder', 'lighter'])
-                || ($mValue instanceof Size
-                    && in_array($mValue->getSize(), range(100, 900, 100)))
+                in_array($mValue, ['bold', 'bolder', 'lighter'], true)
+                // Note: font weights should be integers, not floats. For now, we'll keep using floats. We will
+                // drill down on this and fix it in  https://github.com/MyIntervals/PHP-CSS-Parser/issues/612.
```

Comment:
This comment needs changing (or removing), now that we have closed #612 on the basis that CSS treats all numerical values as a `number`, regardless of whether they may be floating-point.

Reply:
Done.